### PR TITLE
[CPDLP-1552] Show the correct participant role in the delivery partners check data service

### PR DESCRIPTION
**Pull request by leoapost**
### Context

- Ticket: [CPDLP-1552](https://dfedigital.atlassian.net/browse/CPDLP-1552)

The check data service shows wrong participant roles in the UI and csv extract, when users are both Mentors and Induction Tutors.

### Changes proposed in this pull request

- I created the `role` method in the `ParticipantProfile::ECF` model to determine the participant's role based on their participant type
- Replaced the `User#user_description` method that shows the wrong role in both the UI and CSV serializer, with the `ParticipantProfile::ECF#role`

### CODE REVIEW

File: app/models/participant_profile/ecf.rb [hunk 1]
```diff
@@ -79,6 +79,15 @@ def relevant_induction_record(lead_provider:)
         .first
     end
 
+    def role
```
Comment (asmega) [inline app/models/participant_profile/ecf.rb:82]: fyi if you implemented `role` on the subclasses there would be no need for a case statement

File: spec/models/participant_profile/ecf_spec.rb [hunk 2]
```diff
@@ -210,4 +210,22 @@
       end
     end
   end
+
+  describe "#role" do
+    context "when participant is Mentor" do
+      let(:profile) { create(:mentor_participant_profile) }
```
Comment (asmega) [inline spec/models/participant_profile/ecf_spec.rb:216]: i probably would have chosen to use `build` here to do it in memory rather than make the db call

**Review by asmega — APPROVED** [review]

File: app/models/participant_profile/ecf.rb (hunk 1 above)
Comment (leoapost) [reply app/models/participant_profile/ecf.rb]: I moved it in the ParticipantProfile class and overriding it in the subclasses.

File: spec/models/participant_profile/ecf_spec.rb (hunk 2 above)
Comment (leoapost) [reply spec/models/participant_profile/ecf_spec.rb:216]: I went with the `create` because the `build(:mentor_participant_profile)` is failing

**Review by asmega — APPROVED** [review]

**Review by mooktakim — APPROVED** [review]

**Pull request merged**